processing_options.go

```diff
@@ -763,6 +764,19 @@ func applyProcessingOptions(po *processingOptions, options urlOptions) error {
 	return nil
 }
 
+func isAllowedSource(imageURL string) bool {
+	source := strings.Split(imageURL, ":")[0]
+	if conf.AllowedSources == "" {
+		return true
+	}
+	for _, val := range strings.Split(conf.AllowedSources, ",") {
+		if string(val) == source {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
There are a few problems with this code:

1) `source := strings.Split(imageURL, ":")[0]` creates a new slice of strings on every request, which leads to redundant memory allocations. More of all, it will cause panic if `imageURL` doesn't include `:`. Using `strings.HasPrefix` is much more efficient way.
2) `strings.Split(conf.AllowedSources, ",")` creates a new slice of strings on every request as well. It's better to implement a custom config parser that will convert `IMGPROXY_ALLOWED_SOURCES` value from something like `local,s3,gs` to something like `[]string{"local://", "s3://", "gs://"}`, and use this parser on the configuration stage.

Reply:
Implemented custom parser and switched to using HasPrefix